devito/types/sparse.py

```diff
@@ -1067,43 +1064,42 @@ class PrecomputedSparseFunction(AbstractSparseFunction):
 
     _sub_functions = ('gridpoints', 'interpolation_coeffs')
 
-    def __init__(self, *args, **kwargs):
-        if not self._cached():
-            super(PrecomputedSparseFunction, self).__init__(*args, **kwargs)
-
-            # Grid points per sparse point (2 in the case of bilinear and trilinear)
-            r = kwargs.get('r')
-            if not isinstance(r, int):
-                raise TypeError('Need `r` int argument')
-            if r <= 0:
-                raise ValueError('`r` must be > 0')
-            self.r = r
-
-            gridpoints = SubFunction(name="%s_gridpoints" % self.name, dtype=np.int32,
-                                     dimensions=(self.indices[-1], Dimension(name='d')),
-                                     shape=(self.npoint, self.grid.dim), space_order=0,
-                                     parent=self)
-
-            gridpoints_data = kwargs.get('gridpoints', None)
-            assert(gridpoints_data is not None)
-            gridpoints.data[:] = gridpoints_data[:]
-            self._gridpoints = gridpoints
-
-            interpolation_coeffs = SubFunction(name="%s_interpolation_coeffs" % self.name,
-                                               dimensions=(self.indices[-1],
-                                                           Dimension(name='d'),
-                                                           Dimension(name='i')),
-                                               shape=(self.npoint, self.grid.dim,
-                                                      self.r),
-                                               dtype=self.dtype, space_order=0,
-                                               parent=self)
-            coefficients_data = kwargs.get('interpolation_coeffs', None)
-            assert(coefficients_data is not None)
-            interpolation_coeffs.data[:] = coefficients_data[:]
-            self._interpolation_coeffs = interpolation_coeffs
-            warning("Ensure that the provided interpolation coefficient and grid point " +
-                    "values are computed on the final grid that will be used for other " +
-                    "computations.")
+    def __init_finalize__(self, *args, **kwargs):
```

Comment:
note for reviewers: this is exactly like before, but de-indented as we don't need ``if not self._cached()`` anymore